src/librustc_typeck/check/writeback.rs

```diff
@@ -160,8 +162,9 @@ impl<'cx, 'gcx, 'tcx> Visitor<'gcx> for WritebackCx<'cx, 'gcx, 'tcx> {
 
         self.visit_node_id(e.span, e.id);
 
-        if let hir::ExprClosure(_, _, body, _) = e.node {
+        if let hir::ExprClosure(_, _, body, _, _) = e.node {
             let body = self.fcx.tcx.hir.body(body);
+            // FIXME: Why visit the args here?
```

Comment:
The `visit_body` call also visits the node IDs for arguments. The same thing also happens in the `resolve_type_vars_in_body` function.